Formula/v8.rb

```diff
@@ -15,98 +16,211 @@ class V8 < Formula
-    ENV.append "GYP_DEFINES", "v8_use_external_startup_data=0"
-
-    # fix up libv8.dylib install_name
-    # https://github.com/Homebrew/homebrew/issues/36571
-    # https://code.google.com/p/v8/issues/detail?id=3871
-    inreplace "tools/gyp/v8.gyp",
-              "'OTHER_LDFLAGS': ['-dynamiclib', '-all_load']",
-              "\\0, 'DYLIB_INSTALL_NAME_BASE': '#{opt_lib}'"
-
-    (buildpath/"build/gyp").install resource("gyp")
-    (buildpath/"third_party/icu").install resource("icu")
-    (buildpath/"buildtools").install resource("buildtools")
-    (buildpath/"base/trace_event/common").install resource("common")
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    output_path = "out.gn/x64.release"
+
+    # Trick gclient to reuse our v8 checkouts
+    base = "gclient-cache/chromium.googlesource.com"
+    (buildpath/"#{base}-v8-v8").install Dir["*"]
+
+    (resources - [resource("depot_tools")]).each do |r|
+      (buildpath/"#{base}-#{r.name}").install r
+    end
+
+    # Install depot_tools in PATH
+    (buildpath/"depot_tools").install resource("depot_tools")
+    ENV.prepend_path "PATH", buildpath/"depot_tools"
+
+    # Prevent from updating depot_tools on every call
+    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
+    ENV["DEPOT_TOOLS_UPDATE"] = "0"
+
+    # Initialize and sync gclient
+    system "gclient", "root"
+    system "gclient", "config", "--spec", <<~EOS
+      solutions = [
+        {
+          "url": "https://chromium.googlesource.com/v8/v8.git",
+          "managed": False,
+          "name": "v8",
+          "deps_file": "DEPS",
+          "custom_deps": {},
+        },
+      ]
+      target_os = [ "mac" ]
+      target_os_only = True
+    EOS
+
+    system "gclient", "sync",
+      "--cache-dir=#{buildpath}/gclient-cache",
+      "-j", ENV.make_jobs,
+      "-r", version,
+      "--no-history",
+      "-vvv"
+
+    # Enter the v8 checkout
+    cd "v8" do
+      gn_args = {
+        :target_os_only => true,
+        :is_debug => false,
+        :is_component_build => true,
+        :v8_use_external_startup_data => false,
+        :v8_enable_i18n_support => true,
+      }
+
+      # Transform to args string
+      gn_args_string = gn_args.map { |k, v| "#{k}=#{v}" }.join(" ")
+
+      # Build with gn + ninja
+      gen_command = "gn gen \"#{output_path}\" --args=\"#{gn_args_string}\""
+      system gen_command
```

Comment:
`system "gn", "gen", "#{output_path}", "--args=#{gn_args_string}"`
will be clearer, without quoting quotation marks

Reply:
The quotes are required. The args string will have spaces.

Reply:
Calling `system` with multiple args, instead of a single string, will enforce the quoting automatically.

Reply:
I tried it like that, but it didn't work.